secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,229 @@
+//
+//  RecoveryPhraseValidation.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import Foundation
+import ComposableArchitecture
+
+typealias RecoveryPhraseValidationStore = Store<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+typealias RecoveryPhraseValidationViewStore = ViewStore<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+
+struct RecoveryPhraseEnvironment {
+    var mainQueue: AnySchedulerOf<DispatchQueue>
+    var newPhrase: () -> Effect<RecoveryPhrase, AppError>
+}
+
+/// Represents the completion of a group of recovery words by de addition of one word into the given group
+struct RecoveryPhraseStepCompletion: Equatable {
+    var groupIndex: Int
+    var word: String
+}
+
+struct RecoveryPhraseValidationState: Equatable {
+    enum Step: Equatable {
+        case initial
+        case incomplete
+        case complete
+        case valid
+        case invalid
+    }
+
+    static let wordGroupSize = 6
+    static let phraseChunks = 4
+
+    var phrase: RecoveryPhrase
+    var missingIndices: [Int]
+    var missingWordChips: [PhraseChip.Kind]
+    var completion: [RecoveryPhraseStepCompletion]
+
+    var step: Step {
+        guard !completion.isEmpty else {
+            return  .initial
+        }
+
+        guard completion.count >= missingIndices.count else {
+            return .incomplete
+        }
+
+        return .complete
+    }
+
+    var isValid: Bool {
```

Comment:
if valid is already a phase or status of this state, should this not be part of the other computed property that determines the current state from the enum?